src/anamoly_detection/frequency_signatures/abstract_frequency_signature.py

```diff
@@ -0,0 +1,102 @@
+import abc
+from math import ceil, floor
+from collections import deque
+from scapy.all import Packet
+
+
+class AbstractFrequencySignature(abc.ABC):
+    """
+    Abstract class representing any frequency based signature
+    """
+    def __init__(self, equation, dev_equation, window_size=3600, interval_size=600):
+        """
+        Super constructor for implementing classes
+        Initializes the signature using the super class, and it's unique information
+        :param equation: function used to evaluate the average
+        :param dev_equation: function used to evaluate the deviation
+        :param window_size: Size of the window to limit (Limit is applied to this time frame, in seconds)
+        :param interval_size: Segmented window size, sets how often windows are adjusted
+        """
+        # Lambda function equations
+        self._limit_equation = equation
+        self._deviation_equation = dev_equation
+
+        # Frequency for the window as a whole, and for the segments that make up the sliding window
+        self._window_frequency = 0
+        self._interval_frequencies = deque(maxlen=(ceil(window_size / interval_size)))
+        self._last_interval = 0
+
+        # Average and deviation calculated for the current window
+        self._current_average = -1
```

Comment:
Why does it start at -1?

Reply:
Impossible value to reach otherwise, indicates average has yet to be calculated and needs to be (Line 68).